File: java/src/main/java/org/endeavourhealth/library/model/imq/ValueSource.java
package org.endeavourhealth.library.model.imq;


public class ValueSource {
  private String parameter;
  private String iri;
  private String name;
  private String nodeRef;
  private String propertyRef;

  public String getPropertyRef() {
    return propertyRef;
  }

  public ValueSource setPropertyRef(String propertyRef) {
    this.propertyRef = propertyRef;
    return this;
  }


  public String getIri() {
    return iri;
  }

  public ValueSource setIri(String iri) {
    this.iri = iri;
    return this;
  }

  public String getNodeRef() {
    return nodeRef;
  }

  public ValueSource setNodeRef(String nodeRef) {
    this.nodeRef = nodeRef;
    return this;
  }

  public String getName() {
    return name;
  }

  public ValueSource setName(String name) {
    this.name = name;
    return this;
  }



  public String getParameter() {
    return parameter;
  }

  public ValueSource setParameter(String parameter) {
    this.parameter = parameter;
    return this;
  }



}

File: src/interfaces/ValueSource.ts
import z from "zod";

// export interface ValueSource {
//   parameter?: string;
//   iri?: string;
//   name?: string;
//   nodeRef?: string;
//   propertyRef?: string;
// }

export const ValueSourceSchema = z.strictObject({
  parameter: z.string().optional(),
  iri: z.string().optional(),
  name: z.string().optional(),
  nodeRef: z.string().optional(),
  propertyRef: z.string().optional()
});

export type ValueSource = z.output<typeof ValueSourceSchema>;
